doc/source/user_guide/numpy_images.txt

```diff
@@ -55,28 +64,134 @@ Masking (indexing with masks of booleans)::
+    >>> cat.shape
+    (300, 451, 3)
+
+That's a 300-by-451 pixel image with red, green, and blue channels.
+As before, we can get and set pixel values:
+
+    >>> cat[10, 20]
+    array([151, 129, 115], dtype=uint8)
+    >>> # set the pixel at row 50, column 60 to black
+    >>> cat[50, 60] = 0
+    >>> # set the pixel at row 50, column 61 to green
+    >>> cat[50, 61] = [0, 255, 0] # [red, green, blue]
+
+We can also use 2D boolean masks for a 2D color image, as we did with
+the grayscale image above:
+
+.. plot::
+
+    Using a 2D mask on a 2D color image
+
+    >>> reddish = cat[:, :, 0] > 160
+    >>> cat[reddish] = [0, 255, 0]
+    >>> plt.imshow(cat)
+
+
+Coordinate conventions
+----------------------
+
+Because we represent images with numpy arrays, our coordinates must
+match accordingly. Two-dimensional (2D) grayscale images (such as
+`camera` above) are indexed by row and columns (abbreviated to either
+``row, col`` or ``r, c``), with the lowest element (0, 0) at the top-
+-left corner. In various parts of the library, you will
+also see ``rr`` and ``cc`` refer to lists of row and column
+coordinates. We distinguish this from (x, y), which commonly denote
+Cartesian coordinates, where x is the horizontal coordinate, y the
+vertical, and the origin is on the bottom right. (Matplotlib, for
+example, uses this convention.)
+
+In the case of color (or multichannel) images, the last dimension
+contains the color information and is denoted ``channel`` or ``ch``.
+
+Finally, for 3D images, such as videos, magnetic resonance imaging
+(MRI) scans, or confocal microscopy, we refer to the leading dimension
+as ``frame``, abbreviated as ``frm`` or ``f``. In many cases,
+the third imaging dimension has lower resolution than the other two,
+and processing must be done frame-wise. When frames are the leading
+dimension, we can use the following syntax:
+
+    >>> for image in image3d: # iterate over first dimension
+    ...     do_something_to(image)
+
+
+Notes on array order
+--------------------
+
+Although the labeling of the axes seems arbitrary, it can have a
+significant effect on speed of operations. This is because modern
+processors never retrieve just one item from memory, but rather a
+whole chunk of adjacent items. (This is called prefetching.)
+Therefore, processing elements that are
+next to each other in memory is faster than processing them
+in a different order, even if the number of operations is the same:
+
+    >>> def in_order_multiply(arr, scalar):
+    ...     for frame in list(range(arr.shape[0])):
+    ...         arr[frame, :, :] *= scalar
+    ... 
+    >>> def out_of_order_multiply(arr, scalar):
+    ...     for frame in list(range(arr.shape[2])):
+    ...         arr[:, :, frame] *= scalar
+    ... 
+    >>> import time
+    >>> im3d = np.random.rand(100, 1024, 1024)
+    >>> t0 = time.time(); x = in_order_multiply(im3d, 5); t1 = time.time()
+    >>> print("%.2f seconds" % (t1 - t0))
+    0.14 seconds
```

Comment:
Thinking out loud (haven't started writing yet):

Should this script directly modify the existing files? That would be OK, as long as our doc build happens effectively in isolation or in a new branch. However, it has the potential to royally screw up repositories.

The other option would be to copy all of the doc source files to another temp directory, strip out the doctest snippets, build those, and then eliminate the temp directory. This is probably the right call, but it has the potential to be a little messy with a leftover temp directory if the doc build failed in specific ways. 

Thoughts?